jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt

```diff
@@ -306,6 +306,12 @@ class CodeWhispererPopupManager {
         } else {
             WindowManager.getInstance().setAlphaModeRatio(popup.popupWindow, 0.1f)
         }
+        states.requestContext.latencyContext.codewhispererPostprocessingEnd = System.nanoTime()
+        states.requestContext.latencyContext.codewhispererEndToEndEnd = System.nanoTime()
+
+        if (!CodeWhispererInvocationStatus.getInstance().hasExistingInvocation()) {
+            CodeWhispererTelemetryService.getInstance().sendClientComponentLatencyEvent(states)
```

Comment:
Does this guarantee that the code suggestion is shown to the user before client component latency is sent?

Reply:
yes, when is line is executed, the line of showing popup is guranteed to be executed above. But this reminds me an edge case where scrolling would also enter this path and send the event which we don't want. Will fix it.

Reply:
`if (!CodeWhispererInvocationStatus.getInstance().hasExistingInvocation()) {` this is gating for the condition that all paginated response are received.